bin/wasm-node/javascript/src/index.js

```diff
@@ -230,9 +230,17 @@ export function start(config) {
       return chainAddedPromise;
     },
     terminate: () => {
-      worker.terminate();
-      if (!workerError)
-        workerError = new AlreadyDestroyedError();
+      clearTimeout(livenessTimeout)
+
+      // On NodeJS `worker.terminate()` returns a Promise, while on the DOM
+      // it returns `undefined`. In order to have a consistent API we will
+      // always return a Promise
+      const terminationPromise = worker.terminate() || Promise.resolve()
+
+      return terminationPromise.then(() => {
+        if (!workerError)
+          workerError = new AlreadyDestroyedError();
```

Comment:
This needs to be done before we call `worker.terminate()`. The goal is to prevent people from using the `Client` after they have called `terminate()`, which they could still do if you put this in the `then()`.

Reply:
Yeah! That's what I thought, but the previous code was doing it after termination, so I didn't want to change too much... But I will make sure that get's fixed, too. Thanks!

Reply:
done!